### DISCUSSION TITLE
Query Regarding Unaltered Transcripts in Deepgram API STT

### DISCUSSION THREAD

**Discussion by kikyo1012** [deepgram/community#984] [General help] [answered]
Hi Deepgram Support Team,

I have been using the Deepgram API for speech-to-text tasks and specifically require the ASR function to generate transcripts along with timestamps. I've noticed that the transcripts appear slightly polished and differ from the original audio, which seems to cause some alignment issues with the timestamps. Could you guide me on how to adjust the settings so that the transcripts are not altered at all?

Thank you for your assistance.

The script I am using is the following Python code:
`from deepgram import DeepgramClient, PrerecordedOptions, FileSource

def main():
    response_json = None
    try:
        deepgram = DeepgramClient(f"{Deepgram_user}")

        with open(AUDIO_FILE, "rb") as file:
            buffer_data = file.read()

        payload: FileSource = {"buffer": buffer_data,}

        options = PrerecordedOptions(
            model="whisper-large",
            # smart_format=True,
            punctuate=True,
            diarize=True,
            # utt_split= 0.8
        )
        response = deepgram.listen.rest.v("1").transcribe_file(payload, options)

        response_json = response.to_json(indent=4)
        print(response_json)

        with open(f"{base_directory}{video_sample}_generated_audio_ASR_deepgram.json", 'w') as json_file:
            json_file.write(response_json)

    except Exception as e:
        print(f"Exception: {e}")

    return response_json

if __name__ == "__main__":
    main()

with open(f"{base_directory}{video_sample}_generated_audio_ASR_deepgram.json", 'r') as file:
    generated_audio_ASR_deepgram = json.load(file)

`

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment, marked as answer]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- A request ID that triggered your error or issue.

**Reply by kikyo1012** [reply]
"request_id": "bd0c93b1-f7a4-4a86-ab59-5434f15533f9"

**Comment by jkroll-deepgram** [comment]
Hi @kikyo1012, I see you're using the `whisper-large` model. Whisper does modify transcripts to "polish" them. Deepgram transcribes exactly what was said. I recommend trying our `nova-2` model.

**Reply by kikyo1012** [reply]
Thank you so much for  your help. I have a follow-up question: I've noticed that the API often returns an "Exception: The read operation timed out" error when I'm using the `whisper-large` model. Is there anything I can do to reduce these errors?

**Reply by jkroll-deepgram** [reply]
@kikyo1012 Whisper is a very slow model, so that will happen more often on long audio (say, longer than 30-60 minutes or so), as well as with diarization enabled. To reduce that, I recommend breaking long audio into multiple shorter files, and/or disabling diarization (diarize=false).

If you are able to switch from Whisper to Nova-2, you'll see much faster processing times, and this error should not occur.

**Reply by kikyo1012** [reply]
Thank you for the tips! I will definitely switch to the nova-2 model. I've tested it, and it's both faster and more stable. Quick question, if `smart_format=true`, does it polish the transcript in any way (besides dates, times, currency amounts, phone numbers, emails, and URLs.)?

**Reply by jkroll-deepgram** [reply]
@kikyo1012, No, smart formatting does not polish the transcript besides that entity formatting you mentioned, plus punctuation and capitalization. It has high fidelity to exactly what was said.

We also have a [filler words](https://developers.deepgram.com/docs/filler-words) feature for English transcription that you may be interested in, to transcribe what was said even more accurately with "uh", "mhmm", etc. Without the filler words feature enabled, those filler words are omitted from the transcript.

**Comment by kikyo1012** [comment]
Yes, I have noticed these `Formating choices` in the document. Thank you for clarification!
